models/migrations/v1_20/v256.go

```diff
@@ -0,0 +1,23 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package v1_20 //nolint
+
+import (
+	"xorm.io/xorm"
+)
+
+func AddIsInternalColumnToPackage(x *xorm.Engine) error {
+	type Package struct {
+		ID               int64  `xorm:"pk autoincr"`
+		OwnerID          int64  `xorm:"UNIQUE(s) INDEX NOT NULL"`
+		RepoID           int64  `xorm:"INDEX"`
+		Type             string `xorm:"UNIQUE(s) INDEX NOT NULL"`
+		Name             string `xorm:"NOT NULL"`
+		LowerName        string `xorm:"UNIQUE(s) INDEX NOT NULL"`
+		SemverCompatible bool   `xorm:"NOT NULL DEFAULT false"`
+		IsInternal       bool   `xorm:"UNIQUE(s) INDEX NOT NULL DEFAULT false"`
```

Comment:
Does https://github.com/go-gitea/gitea/pull/22854#discussion_r1180809086 answer your question?

Reply:
No. If `LowerName` and `IsInternal` is used together, then the index of `IsInternal` is never used.

Reply:
And I think the `INDEX` on `LowerName` is also unnecessary, because it already has `UNIQUE`, it is also an index.

Reply:
However, it's a combined index which as I've learned means it won't be included as it isn't the first column of the index.

Reply:
IIRC, in XORM , if we write `UNIQUE(s)`, then there will be an unique index

But if we add the extra `INDEX`, there will be a new/individual index for it.

Reply:
> And I think the `INDEX` on `LowerName` is also unnecessary, because it already has `UNIQUE`, it is also an index.

I was today old when I learned xorm treats both different. Everywhere it's just called `UNIQUE INDEX` which would result in one index. It is not intended to create a unique index AND an individual index for every field.

For example:
```go
// models/user/redirect.go
type Redirect struct {
	ID             int64  `xorm:"pk autoincr"`
	LowerName      string `xorm:"UNIQUE(s) INDEX NOT NULL"`
	RedirectUserID int64  // userID to redirect to
}
```
That does not make sense.

Reply:
That's just wrong.

![image](https://user-images.githubusercontent.com/2114189/235357882-5d795b68-c8db-41b4-a38b-89cd6d88df59.png)

Reply:
Is this resolved?